chore: adding API Key guides and suggestions (#77)
Basis-Theory/terraform-provider-basistheory: docs/resources/application.md

### Optional

- `create_key` (Boolean) Create key by default for the Application. Do not set to 'true' if you want to manage the key with the 'basistheory_application_key' resource

>   - We strongly suggest you leave the `create_key` default for new Applications, for existing Applications read our guide on [migrating to v2](/docs/guides/v1_v2_migration.md#basistheory_application-create_key-and-basistheory_application_keys) 

Reviewer: ```suggestion
  - We strongly suggest you to omit `create_key` in new Applications. For existing Applications read our guide on [migrating to v2](/docs/guides/v1_v2_migration.md#basistheory_application-create_key-and-basistheory_application_keys) 
```
